src/main/java/org/wildfly/security/http/util/sso/DefaultSingleSignOnSession.java

```diff
@@ -109,32 +109,33 @@ public void put(SecurityIdentity identity) {
             scope.registerForNotification(notification -> {
                 HttpScope sessionScope = notification.getScope(Scope.SESSION);
 
-                Collection<Map.Entry<String, URI>> logoutTargets = Collections.emptyList();
-                try (SingleSignOn target = this.context.getSingleSignOnManagerManager().find(id)) {
+                Map<String, Map.Entry<String, URI>> logoutTargets = Collections.emptyMap();
+                try (SingleSignOn target = this.context.getSingleSignOnManager().find(id)) {
                     if (target != null) {
                         Map.Entry<String, URI> localParticipant = target.removeParticipant(applicationId);
                         if (localParticipant != null) {
                             log.debugf("Removed local session [%s] from SSO [%s]", localParticipant.getKey(), target.getId());
                         }
                         if (sessionScope.getAttachment(SESSION_INVALIDATING_ATTRIBUTE) == null) {
-                            Collection<Map.Entry<String, URI>> participants = target.getParticipants();
+                            Map<String, Map.Entry<String, URI>> participants = target.getParticipants();
                             if (participants.isEmpty()) {
                                 log.debugf("Destroying SSO [%s]. SSO is not associated with participants", target.getId());
                                 target.invalidate();
                             } else if (notification.isOfType(INVALIDATED)) {
-                                logoutTargets = participants;
+                                logoutTargets = new HashMap<>(participants);
```

Comment:
Is this necessary? The result from SingleSignOn.getParticipants() is already unmodifiable.  If the backing map were not concurrent (it is), cloning the map would still be vulnerable to ConcurrentModificationExceptions.